iceoryx_posh/include/iceoryx_posh/popo/base_subscriber.hpp

```diff
@@ -83,31 +83,49 @@ class BaseSubscriber
     ///
     void unsubscribe() noexcept;
 
+    // iox-#408 replace
     ///
     /// @brief hasData Check if sample is available.
     /// @return True if a new sample is available.
     ///
     bool hasSamples() const noexcept;
```

Comment:
do you really want to completely replace this? This is also useful for the typed publisher

Reply:
I was thinking about having it called chunks in the base and untyped and samples in the typed class. Or just have both in the base and pull in the one needed in the derived class.